curriculum/challenges/english/25-front-end-development/lab-house-painting/66d6a7a3e1aa411e94bf2346.md

```diff
@@ -0,0 +1,532 @@
+```
+
+You should target `#door` and set its `border` property.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle("#door")?.getPropertyValue("border"));
+```
+
+You should target `#door` and set its `background-color` property.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle("#door")?.getPropertyValue("background-color"));
+```
+
+Your `#roof` should have the `top` property set to `0`.
+
+```js
+assert.equal(getComputedStyle(document.querySelector("#roof"))?.getPropertyValue("top"), "0px");
+```
+
+Your `#door` should be placed at the bottom of the house.
+
+```js
+const door = getComputedStyle(document.querySelector("#door"))
+const doorBottom = Number(door.getPropertyValue("bottom").replace("px", ""));
+assert.strictEqual(doorBottom, 0);
+```
+
+Your `#door` should not be outside the house borders.
+
+```js
+const door = getComputedStyle(document.querySelector("#door"))
+const doorLeft = Number(door.getPropertyValue("left").replace("px", ""));
+const doorRight = Number(door.getPropertyValue("right").replace("px", ""));
+assert.isAbove(doorLeft, 0);
+assert.isAbove(doorRight, 0);
+```
+
+Your `#chimney` should have a `top` value that puts it at the top of your `#house`.
+
+```js
+const chimney = getComputedStyle(document.querySelector("#chimney"));
+assert.equal(Number(chimney?.getPropertyValue("top").replace("px", "")), - Number(chimney?.getPropertyValue("height").replace("px", "")));
+```
+
+Your `#chimney` should have a `z-index` that puts it behind the house.
+
+```js
+const houseZIndex = getComputedStyle(document.querySelector("#house"))?.getPropertyValue("z-index");
+const chimneyZIndex = getComputedStyle(document.querySelector("#chimney"))?.getPropertyValue("z-index");
+if (houseZIndex === "auto") {
+    assert.isBelow(Number(chimneyZIndex), 0)
+} else {
+    assert.isBelow(Number(chimneyZIndex), Number(houseZIndex));
+}
+```
+
+Your windows should be placed below your `#roof` and at least higher than one third of your `#door`'s height.
+
+```js
+const window1 = getComputedStyle(document.querySelector("#window-1"))
+const window2 = getComputedStyle(document.querySelector("#window-2"))
+const door = getComputedStyle(document.querySelector("#door"))
+const roof = getComputedStyle(document.querySelector("#roof"))
+const w1Top = Number(window1.getPropertyValue("top").replace("px", ""));
+const w1Bottom = Number(window1.getPropertyValue("bottom").replace("px", ""));
+const w2Top = Number(window2.getPropertyValue("top").replace("px", ""));
+const w2Bottom = Number(window2.getPropertyValue("bottom").replace("px", ""));
+const doorHeight = Number(door.getPropertyValue("height").replace("px", ""));
+const roofHeight = Number(roof.getPropertyValue("height").replace("px", ""));
+assert.isAbove(w1Top, roofHeight);
+assert.isAbove(w1Bottom, doorHeight/3);
+assert.isAbove(w2Top, roofHeight);
+assert.isAbove(w2Bottom, doorHeight/3);
+```
+
+Your windows should not be outside the house borders.
```

Comment:
Can this hint be elaborated in the instruction?

I couldn't figure it out from the hint what was required and the test failed even when the window `div` were inside the house like this:

![image](https://github.com/user-attachments/assets/669f930f-078b-481f-aeba-6a3119dbeacc)